clients/privacy-center/packages/fides-consent/src/lib/consent-value.ts

```diff
@@ -0,0 +1,43 @@
+import { ConsentContext } from "./consent-context";
+
+export type ConditionalValue = {
+  value: boolean;
+  globalPrivacyControl: boolean;
+};
+
+/**
+ * A consent value can be a boolean:
+ *  - `true`: consent/opt-in
+ *  - `false`: revoke/opt-out
+ *
+ * A consent value can also be context-dependent, which means it will be decided based on
+ * information about the user's environment (browser). The `ConditionalValue` object maps the
+ * context conditions to the value that should be used:
+ *  - `value`: The default value if no context applies.
+ *  - `globalPrivacyControl`: The value to use if the user's browser has Global Privacy Control
+ *    enabled.
+ */
+export type ConsentValue = boolean | ConditionalValue;
+
+export type ConsentTypeToValue = {
+  [consentType: string]: ConsentValue;
+};
+
+export const resolveConsentValue = (
+  value: ConsentValue | undefined,
+  context: ConsentContext
+): boolean => {
+  if (value === undefined) {
+    return false;
+  }
+
+  if (typeof value === "boolean") {
+    return value;
+  }
+
+  if (context.globalPrivacyControl === true) {
```

Comment:
I'm confused about what this means.

My understanding of what it should be - If the GPC is enabled by the user and GPC is enabled for the consent option, then we set the value of the consent option to False (opted-out).
What's happening here -  If the GPC is enabled by the user, then we set the value of the consent option to if GPC applies to the consent or not. 

Example - So let's say the user has GPC as true (User has opted out) and in the config `consentValue.globalPrivacyControl ` is true. Then we set the consent value to be true (the user has opted in) but actually we should set it to false or opted-out.

My JS knowledge is a bit rusty please excuse me if what I'm saying doesn't make sense.

Reply:
In addition to our other thread, I've updated the readme(s) to describe how this works:
https://github.com/ethyca/fides/blob/93098f0bf4162117a5842e8fdc300b970ffc4471/clients/privacy-center/packages/fides-consent/README.md#consent-context